agent/src/capability/baseconfigstore.cpp

```diff
@@ -23,173 +23,201 @@ namespace softwarecontainer {
+void BaseConfigStore::readCapsFromDir(const std::string &dirPath)
 {
+    std::string errorMessage;
+
     if (dirPath.compare("/") == 0) {
-        log_warning() << "Searching for configuration files from root dir not allowed";
-        return ReturnCode::FAILURE;
+        errorMessage = "Searching for configuration files from root dir not allowed";
+        log_warning() << errorMessage;
+        throw ServiceManifestPathError(errorMessage);
     }
     if (isDirectoryEmpty(dirPath)) {
         log_warning() << "Path to configuration files is empty: " << dirPath;
-        return ReturnCode::SUCCESS;
+        return;
     }
 
     std::vector<std::string> files = fileList(dirPath);
 
     if (files.empty()) {
         log_info() << "No configuration files found: " << dirPath;
-        return ReturnCode::SUCCESS;
+        return;
     }
 
     for (std::string file : files) {
         std::string filePath = dirPath + file;
 
-        if (isError(readCapsFromFile(filePath))) {
-            log_warning() << "Could not parse a file in directory: " << filePath;
-            return ReturnCode::FAILURE;
+        try {
+            readCapsFromFile(filePath);
+        } catch(ConfigStoreError &error) {
+            errorMessage = "Could not parse a file in directory";
+            log_warning() << errorMessage << ": " << filePath;
+            throw ServiceManifestPathError(errorMessage);
         }
     }
 
-    return ReturnCode::SUCCESS;
+    //return;
 }
 
-ReturnCode BaseConfigStore::readCapsFromFile(const std::string &filePath)
+void BaseConfigStore::readCapsFromFile(const std::string &filePath)
 {
+    std::string errorMessage;
+
     if (!isJsonFile(filePath)) {
-        log_debug() << "File is not a json file: " << filePath;
-        return ReturnCode::FAILURE;
+        errorMessage = "File is not a json file";
+        log_debug() << errorMessage << ": " << filePath;
+        throw ServiceManifestPathError(errorMessage);
     }
 
     json_error_t error;
     json_t *fileroot = json_load_file(filePath.c_str(), 0, &error);
 
     if (nullptr == fileroot) {
-        log_error() << "Could not parse Service Manifest: "
-                    << filePath << ":" << error.line <<" : " << error.text;
-        return ReturnCode::FAILURE;
+        errorMessage = "Could not parse Service Manifest";
+        log_error() << errorMessage << ": " << filePath << ":" << error.line <<" : " << error.text;
+        throw ServiceManifestPathError(errorMessage);
     } else if (!json_is_object(fileroot)) {
-        log_error() << "Configuration is not a json object: \n"
-                    << filePath << ":" << error.line <<" : " << error.text;
-        return ReturnCode::FAILURE;
+        errorMessage = "Configuration is not a json object";
+        log_error() << errorMessage << ": " << filePath << ":" << error.line <<" : " << error.text;
+        throw ServiceManifestPathError("Service Manifest parse error - " + errorMessage);
```

Comment:
I guess this comment is about the message repeating the type of error? In this case it's repeated since it's in the errorMessage as well as the prefixed string, but I think it has its uses to repeat the info that the type would imply. If we catch the error on one of its base types it would be good to be able to generate useful log messages from it without knowing the type.